asapdiscovery/simulation/scripts/MProFunction.py

```diff
@@ -0,0 +1,239 @@
+hydrogenMass = 4.0*unit.amu
+platform = Platform.getPlatformByName('CUDA')
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+
+    ## Input arguments
+    parser.add_argument(
+        "-i",
+        "--input_dir",
+        type = str,
+        default = "/data/chodera/asap-datasets/current/sars_01_prepped_v3/",
+        help = "Path to PDB and SDF files"
+    )
+    parser.add_argument(
+        "-p", 
+        "--protein", 
+        type = str,
+        default = "/data/chodera/lemonsk/asap-datasets/openmm_setup_processed/prepped_receptor_0-processed.pdb",
+        help = "Path to PDB file to simulate"
+    )
+    parser.add_argument(
+        "-l", 
+        "--ligand",
+        type = str,
+        default = "/data/chodera/lemonsk/covid-moonshot-ml/asapdiscovery/simulation/tests/inputs/MAT-POS-f2460aef-1.sdf",
+        help="Ligand"
+    )
+    parser.add_argument(
+        "-n",
+        "--nsteps",
+        type = int,
+        default = 12,
+        help = "Number of Steps"
+    )
+    parser.add_argument(
+        "-o",
+        "--output_dir",
+        type = str,
+        default = "/data/chodera/lemonsk/asap-datasets/MPro_Simulations/",
+        help = "Output simulation directory."
+    )
+    args = parser.parse_args()
+    return args
+
+
+def simulate_mpro(input_dir,protein,ligand,nsteps,output_dir):
+    ## Necessary for Reporters
+    save = re.search(r"Mpro-P+\d{4}", protein).group(0)
+    logfile = output_dir / f"{save}-log.txt"
+    logger = logging.getLogger(save)
+    logger.setLevel(logging.DEBUG)
+    handler = logging.FileHandler(str(logfile), mode="w")
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter("%(asctime)s | %(levelname)s | %(message)s")
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    num_steps = int(nsteps) # number of integrator steps
+    n_snapshots = int(num_steps / reportingInterval) # calculate number of snapshots that will be generated
+    num_steps = n_snapshots * reportingInterval # recalculate number of steps to run
+
+    # Read the molfile into RDKit, add Hs and create an openforcefield Molecule object
+    logger.info(':pill:  Reading ligand')
+    rdkitmol = Chem.SDMolSupplier(ligand)[0]
+    logger.info(f':mage:  Adding hydrogens')
+    rdkitmolh = Chem.AddHs(rdkitmol, addCoords=True)
+    # ensure the chiral centers are all defined
+    Chem.AssignAtomChiralTagsFromStructure(rdkitmolh)
+    ligand_mol = Molecule(rdkitmolh)
+
+    # Initialize a SystemGenerator
+    logger.info(':factory: System Combination')
+    forcefield_kwargs = {'constraints': app.HBonds, 'rigidWater': True, 'removeCMMotion': False, 'hydrogenMass': 4*unit.amu }
+    periodic_forcefield_kwargs = {'nonbondedMethod': app.PME}
+    system_generator = SystemGenerator(
+        forcefields=['amber/ff14SB.xml', 'amber/tip3p_standard.xml'],
+        small_molecule_forcefield='openff-1.3.1',
+        molecules=[ligand_mol], cache='cache.json',
```

Comment:
you define simulations parameters at the beginning but then you don't use them here. You should use them!

Reply:
Ah I see. This bit is from Jenke's script, I will go in and remove some fluff